subsystem/metrics_test.go

```diff
@@ -0,0 +1,276 @@
+	var validationRes map[string][]hostValidationResult
+	h := getHost(clusterID, hostID)
+	if h.ValidationsInfo == "" {
+		return false, nil
+	}
+	err := json.Unmarshal([]byte(h.ValidationsInfo), &validationRes)
+	Expect(err).ShouldNot(HaveOccurred())
+	for _, vRes := range validationRes {
+		for _, v := range vRes {
+			if v.ID != validationID {
+				continue
+			}
+			return v.Status == "failure", nil
+		}
+	}
+	return false, nil
+}
+
+func filterMetrics(metrics []string, substrings ...string) []string {
+	var res []string
+	for _, m := range metrics {
+		containsAll := true
+		for _, ss := range substrings {
+			if !strings.Contains(m, ss) {
+				containsAll = false
+				break
+			}
+		}
+		if containsAll {
+			res = append(res, m)
+		}
+	}
+	return res
+}
+
+func assertValidationCounter(clusterID strfmt.UUID, validationID models.HostValidationID, expectedCounter int) {
+
+	url := fmt.Sprintf("http://%s/metrics", Options.InventoryHost)
+
+	cmd := exec.Command("curl", "-s", url)
+	output, err := cmd.Output()
+	Expect(err).NotTo(HaveOccurred())
+
+	metrics := strings.Split(string(output), "\n")
+	filteredMetrics := filterMetrics(metrics, string(clusterID), hostValidationMetric, string(validationID))
+	Expect(len(filteredMetrics)).To(Equal(1))
+
+	counter, err := strconv.Atoi(strings.ReplaceAll((strings.Split(filteredMetrics[0], "}")[1]), " ", ""))
+	Expect(err).NotTo(HaveOccurred())
+	Expect(counter).To(Equal(expectedCounter))
+}
+
+func metricsDeregisterCluster(ctx context.Context, clusterID strfmt.UUID) {
+
+	_, err := userBMClient.Installer.DeregisterCluster(ctx, &installer.DeregisterClusterParams{
+		ClusterID: clusterID,
+	})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+var _ = Describe("Metrics tests", func() {
+
+	var (
+		ctx       context.Context = context.Background()
+		clusterID strfmt.UUID
+	)
+
+	BeforeEach(func() {
+		cluster, err := userBMClient.Installer.RegisterCluster(ctx, &installer.RegisterClusterParams{
+			NewClusterParams: &models.ClusterCreateParams{
+				Name:             swag.String("test-metrics-cluster"),
+				OpenshiftVersion: swag.String(common.DefaultTestOpenShiftVersion),
+				PullSecret:       swag.String(pullSecret),
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		clusterID = *cluster.GetPayload().ID
+	})
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	Context("Host validation metrics", func() {
+
+		It("connected", func() {
+		})
+
+		It("has-inventory", func() {
+		})
+
+		//FIXME: uncomment commented part to add disk validation test
+		It("has-min-hw-capacity", func() {
+
+			host := &registerHost(clusterID).Host
+
+			var lowHwInfo models.Inventory
+			err := copier.CopyWithOption(&lowHwInfo, validHwInfo, copier.Option{DeepCopy: true})
+			Expect(err).NotTo(HaveOccurred())
+
+			lowHwInfo.CPU.Count = 1
+			lowHwInfo.Memory.PhysicalBytes = int64(4 * units.GiB)
+			//lowHwInfo.Disks = []*models.Disk{
```

Comment:
Please clear comments in this file if they are not needed.

Reply:
This will be uncomment in this task https://issues.redhat.com/browse/MGMT-3645.
The reasons it is there now is so I don't forget it.
The reason it is not uncommented on this PR is because it discovered several bugs and merging it will make the CI fail.

https://issues.redhat.com/browse/MGMT-3630
https://issues.redhat.com/browse/MGMT-3648

Do you still think I should remove it? @filanov

Reply:
If it's going to be used then mayeb you should add TODO. otherwise it is just garbage

Reply:
There is a `FIXME` at the beginning of that `It`